routing/testdataclient/dataclient.go

```diff
@@ -121,3 +127,7 @@ func (c *Client) UpdateDoc(upsertDoc string, deletedIDs []string) error {
 func (c *Client) FailNext() {
 	c.failNext++
 }
+
+func (c *Client) Close() {
+	close(c.quit)
```

Comment:
Maybe wrap with `sync.Once` as I started to do on many (hopefully most/all) other `Close()` ?

Reply:
This is a testdataclient so it might be better to fail on double close.